pkg/controllermanager/controller/factory.go

```diff
@@ -127,7 +127,7 @@ func (f *GardenControllerFactory) Run(ctx context.Context) {
 		projectController                = projectcontroller.NewProjectController(f.clientMap, f.k8sGardenCoreInformers, f.k8sInformers, f.cfg, f.recorder)
 		secretBindingController          = secretbindingcontroller.NewSecretBindingController(f.clientMap, f.k8sGardenCoreInformers, f.k8sInformers, f.recorder)
 		seedController                   = seedcontroller.NewSeedController(f.clientMap, f.k8sGardenCoreInformers, f.k8sInformers, f.cfg, f.recorder)
-		shootController                  = shootcontroller.NewShootController(f.clientMap, f.k8sGardenCoreInformers, f.k8sInformers, f.cfg, f.recorder)
+		shootController                  = shootcontroller.NewShootController(k8sGardenClient, f.k8sGardenCoreInformers, f.k8sInformers, f.cfg, f.recorder)
```

Comment:
Imo, it's confusing that the GCM deals with the client map at many places although the only client it can really get is the garden client, for other clients it will always error. Passing the garden client directly not only reduces the error handling but also simplifies the usage within the controller functions. I'd additionally vote to do the same for alle GCM controllers.

Reply:
Well, the Plant controller also uses the client map for retrieving a Plant client. Also the client map is responsible for refreshing the client's server version (used in some charts to check which API versions should be used).
IISIC, gcm doesn't depend on this in its charts, but that's an easy point to miss.